sched/semaphore/spinlock.c

```diff
@@ -189,6 +224,25 @@ spinlock_t spin_trylock(FAR volatile spinlock_t *lock)
 
 spinlock_t spin_trylock_wo_note(FAR volatile spinlock_t *lock)
 {
+#if defined(CONFIG_TICKET_SPINLOCK)
+
+  if (spin_islocked(lock))
+    {
+      return SP_LOCKED;
+    }
+
+  ticket_spinlock_t old = atomic_load((ticket_spinlock_t *)lock);
```

Comment:
```
#ifdef CONFIG_TICKET_SPINLOCK
uint16_t ticket = atomic_load(&lock->next);
spinlock_t old =
{
  ticket, ticket
};
spinlock_t new =
{
  ticket, ticket + 1
};

if (atomic_compare_exchange_strong(lock.value, &old.value, new.value))
#else
if (up_testset(lock) == SP_LOCKED)
#endif
```

Reply:
I miss the problem. Thank you for your guidance.